packages/snaps-execution-environments/scripts/build.js

```diff
@@ -0,0 +1,197 @@
+/* eslint-disable no-console, node/global-require, node/no-process-exit */
+const {
+  createResolvePath,
+} = require('babel-plugin-tsconfig-paths-module-resolver');
+const browserify = require('browserify');
+const { promises: fs } = require('fs');
+const LavaMoatBrowserify = require('lavamoat-browserify');
+const path = require('path');
+const yargs = require('yargs');
+
+const defaultResolvePath = createResolvePath();
+
+const ENTRY_POINTS = {
+  iframe: { entryPoint: './src/iframe/index.ts', html: true, node: false },
+  offscreen: {
+    entryPoint: './src/offscreen/index.ts',
+    html: true,
+    node: false,
+  },
+  'node-thread': {
+    entryPoint: './src/node-thread/index.ts',
+    html: false,
+    node: true,
+  },
+  'node-process': {
+    entryPoint: './src/node-process/index.ts',
+    html: false,
+    node: true,
+  },
+};
+const OUTPUT_PATH = './dist/browserify';
+const OUTPUT_HTML = 'index.html';
+const OUTPUT_BUNDLE = 'bundle.js';
+
+/**
+ * Builds snaps execution environments using Browserify and LavaMoat.
+ *
+ */
+async function main() {
+  const {
+    argv: { writeAutoPolicy },
+  } = yargs(process.argv.slice(2)).usage(
+    '$0 [options]',
+    'Build snaps execution environments',
+    (yargsInstance) =>
+      yargsInstance.option('writeAutoPolicy', {
+        alias: ['p'],
+        default: false,
+        demandOption: false,
+        description: 'Whether to regenerate the LavaMoat policy or not',
+        type: 'boolean',
+      }),
+  );
+
+  await Promise.all(
+    Object.entries(ENTRY_POINTS).map(async ([key, config]) => {
+      console.log('Bundling', key);
+      const { html, entryPoint, node } = config;
```

Comment:
(nitpick) html and node are not the best names for these values going forward. could indicate where they're coming from.